test/irmin-pack/test_inode.ml

```diff
@@ -51,12 +51,88 @@ module H_contents =
       let t = Irmin.Type.string
     end)
 
+module StepMap = struct
+  include Map.Make (struct
+    type t = Path.step
+
+    let compare = Irmin.Type.(unstage (compare Path.step_t))
+  end)
+
+  let of_list l = List.fold_left (fun acc (k, v) -> add k v acc) empty l
+end
+
+module StepSet = struct
+  include Set.Make (struct
+    type t = Path.step
+
+    let compare = compare
+  end)
+
+  let of_list l = List.fold_left (fun acc v -> add v acc) empty l
+
+  let t =
+    let open Irmin.Type in
+    map (list Path.step_t) of_list elements
+end
+
+module StepSetMap = Map.Make (struct
+  type t = StepSet.t
+
+  let compare = Irmin.Type.(unstage (compare StepSet.t))
+end)
+
 let normal x = `Contents (x, Metadata.default)
 let foo = H_contents.hash "foo"
 let bar = H_contents.hash "bar"
 let check_hash = Alcotest.check_repr Inode.Val.hash_t
 let check_values = Alcotest.check_repr Inode.Val.t
 
+(** Use brute force to generate a [step] from a list of index that this [step]
+    should map to through the [Private.index] function. *)
+let gen_step : int list -> Path.step =
+  let tbl = Hashtbl.create 10 in
+  let max_brute_force_iterations = 100 in
+  fun indices ->
+    let step_of_small_int i =
```

Comment:
maybe use the [random functions from `common.ml`](https://github.com/mirage/irmin/blob/master/test/irmin-pack/common.ml#L20)

Reply:
One of the benefits of `gen_step` is that that the output step is deterministically computed from the input list of int. Thanks to that feature it is easier to compare two test cases, as their steps share a common meaning, i.e. the location in the tree.

Reply:
But I added a caching mechanism and the rng is seeded, so my argument doesn't hold

Reply:
I have added a `val random_letters : int -> string` to commons, to keep printable chars.